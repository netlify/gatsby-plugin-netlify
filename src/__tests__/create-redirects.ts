import { readFile } from 'fs-extra'

import createRedirects from '../create-redirects'

import { createPluginData } from './helpers'

jest.mock(`fs-extra`, () => {
  const actualFsExtra = jest.requireActual(`fs-extra`)
  return {
    ...actualFsExtra,
  }
})

const redirects = [
  {
    "fromPath": "/foo",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/bar"
  },
  {
    "fromPath": "/blog",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/canada/blog",
    "conditions": {
      "country": "ca"
    }
  },
  {
    "fromPath": "/speaker",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/english/speaker",
    "conditions": {
      "language": [
        "en"
      ]
    }
  },
  {
    "fromPath": "/speaker",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/french/speaker",
    "conditions": {
      "language": [
        "fr"
      ]
    }
  },
  {
    "fromPath": "/param?id=7",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/test"
  },
  {
    "fromPath": "/dogs/*",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/animals/*"
  },
  {
    "fromPath": "/cats?id=:id",
    "isPermanent": false,
    "ignoreCase": true,
    "redirectInBrowser": false,
    "toPath": "/animals/:id"
  }
]

describe(`create-redirects`, () => {

  it(`writes file`, async () => {
    const pluginData = await createPluginData()

    await createRedirects(pluginData, redirects, [])

    const output = await readFile(pluginData.publicFolder(`_redirects`), `utf8`)
    expect(output).toMatchSnapshot()
  })
})
