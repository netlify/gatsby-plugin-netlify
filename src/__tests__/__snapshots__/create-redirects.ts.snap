// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-redirects writes file 1`] = `
"
## Created with gatsby-plugin-netlify
/foo  /bar  302
/blog  /canada/blog  302  Country=ca
/speaker  /english/speaker  302  Language=en
/speaker  /french/speaker  302  Language=fr
/param id=7  /test  302
/dogs/*  /animals/:splat  302
/cats id=:id  /animals/:id  302"
`;
